readme.MD

```diff
@@ -1,300 +1,195 @@
+*   **`into table <columns...>`**: Transforms the data into a table with the specified columns.
+*   **`into summary <columns...> calculate <aggregates...>`**: Groups data by the specified columns and calculates aggregate statistics (like counts or sums).
 
-Expression:
-* `f.` functionname 
-* `func.` functionname 
-* `f.` functionname `[` function-arguments... `]`
-* `func.` functionname `[` function-arguments... `]`
-* `h.` headername
-* `header.` headername
-* `p.` propertyname
-* `property.` propertyname
-* `c.` columnname
-* `column.` columnname
-* string
+### Expressions & Functions
 
-headername:
-* TEXT
+You can refer to data fields and transform them using functions:
 
-functionname:
-* TEXT
+*   **Columns/Headers**:
+    *   `c.columnName` (CSV/Table)
+    *   `h.HeaderName` (Mail)
+    *   `p.PropertyName` (iCal)
+*   **Literals**: Strings starting with a dot (e.g., `.gmail`) are treated as text values.
+*   **Functions**: Prefixed with `f.` (e.g., `f.count`, `f.year[c.date]`).
 
-propertyname:
-* TEXT
+**Common Functions:**
 
-string:
-* `.` TEXT
+| Function | Description |
+| :--- | :--- |
+| `f.count` | Counts the number of items in a group. |
+| `f.sum[field]` | Sums the values of a numeric field. |
+| `f.year[date]` | Extracts the year from a date string or timestamp. |
+| `f.month[date]` | Extracts the month from a date string or timestamp. |
+| `f.as[expr, name]` | Renames a column (e.g., `f.as[h.subject, .Title]`). |
 
-## Notes:
-* Certain "expressions" are only available for certain data types:
-  * `column` CSV, table, group, data
-  * `header` Mail
-  * `property` ICal
+See [functions.md](functions.md) for a complete list.
 
-More is definitely required, in most cases it shouldn't be too hard to add. That or create an issue, or if it's too hard
-tell me why in an issue / discussion.
+---
 
-## Examples:
+## Examples
 
-### Summarise senders in a mailbox
+### 1. CSVTrace: Analyze Expenses
 
+Imagine a CSV `expenses.csv` with columns: `Date`, `Category`, `Amount`.
+
+**Task:** Show me all "Food" expenses, showing just the Date and Amount.
 ```bash
-mailtrace -input inbox.mbox -input-type mbox -parser basic \
-    into summary header.from calculate f.count
+csvtrace -input expenses.csv -parser basic \
+  filter c.Category icontains .Food \
+  into table c.Date c.Amount
 ```
 
-### Count calendar events per month
-
+**Task:** How much did I spend on each category?
 ```bash
-icaltrace -input calendar.ics -parser basic \
-    into summary property.start calculate f.count f.month[property.start]
+csvtrace -input expenses.csv -parser basic \
+  into summary c.Category calculate f.sum[c.Amount]
 ```
-
+*Output:*
+```text
++----------+------------+
+| CATEGORY | SUM_AMOUNT |
++----------+------------+
+| Food     |     150.50 |
+| Transport|      45.00 |
+| Utilities|     120.00 |
++----------+------------+
 ```
-go run ./cmd/csvtrace -input ast/testdata/data10.csv -input-type csv -output-type table -parser basic
```

Comment:
@jules please find a use for all these examples even if it means simplifying them